bigtable/src/main/scala/magnolify/bigtable/BigtableType.scala

```diff
@@ -219,14 +224,66 @@ object BigtableField {
     ByteString.copyFrom(bb.putInt(scale).put(unscaled).array())
   }
 
-  implicit def btfOption[A](implicit btf: BigtableField[A]): BigtableField[Option[A]] =
-    new BigtableField[Option[A]] {
-      override def get(xs: java.util.List[Column], k: String)(cm: CaseMapper): Value[Option[A]] =
+  implicit def btfOption[T](implicit btf: BigtableField[T]): BigtableField[Option[T]] =
+    new BigtableField[Option[T]] {
+      override def get(xs: java.util.List[Column], k: String)(cm: CaseMapper): Value[Option[T]] =
         Columns.findNullable(xs, k).map(btf.get(_, k)(cm).toOption).getOrElse(Value.Default(None))
 
-      override def put(k: String, v: Option[A])(cm: CaseMapper): Seq[SetCell.Builder] =
+      override def put(k: String, v: Option[T])(cm: CaseMapper): Seq[SetCell.Builder] =
         v.toSeq.flatMap(btf.put(k, _)(cm))
     }
+
+  implicit def btfIterable[T, C[T]](implicit
+    btf: Primitive[T],
+    ti: C[T] => Iterable[T],
+    fc: FactoryCompat[T, C[T]]
+  ): Primitive[C[T]] =
+    new Primitive[C[T]] {
+      override val size: Option[Int] = None
+
+      override def fromByteString(v: ByteString): C[T] = {
+        val buf = v.asReadOnlyByteBuffer()
+        val n = buf.getInt
+        val b = fc.newBuilder
+        btf.size match {
+          case Some(size) =>
+            val ba = new Array[Byte](size)
+            (1 to n).foreach { _ =>
+              buf.get(ba)
+              b += btf.fromByteString(ByteString.copyFrom(ba))
+            }
+          case None =>
+            (1 to n).foreach { _ =>
+              val s = buf.getInt
+              val ba = new Array[Byte](s)
```

Comment:
I wonder if we need to handle the case with an empty string somehow explicitly, but you have added a test for `Repeated` so looks like it should work.

Reply:
Yeah, arrays can be empty, and scalacheck should generate "" cases.